front/src/applications/stdcm/views/OSRDCStdcmConfig.tsx

```diff
@@ -25,7 +25,9 @@ import RunningTime from '../components/RunningTime';
 
 type OSRDStdcmConfigProps = {
   currentStdcmRequestStatus: string;
-  setCurrentStdcmRequestStatus: (status: string) => void;
+  setCurrentStdcmRequestStatus: React.Dispatch<
+    React.SetStateAction<'IDLE' | 'PENDING' | 'SUCCESS' | 'REJECTED' | 'CANCELED' | 'NORESULTS'>
+  >;
```

Comment:
To make the type more maintainable, you could create this type in const.ts in operational studies folder :
`export type StdcmRequestStatusValues =
    (typeof STDCM_REQUEST_STATUS)[keyof typeof STDCM_REQUEST_STATUS];
`

and then transform the setCurrentStdcmRequestStatus
```suggestion
  setCurrentStdcmRequestStatus: (stdcmRequestStatus: StdcmRequestStatusValues) => void;
```:

Reply:
You can also remove the propTypes and defaultProps in this file between line 148 and 153

Reply:
good, but I think it's better to create a const.ts file in SDTCM as it's a different application